authorization/authorization_integration_test.go

```diff
@@ -0,0 +1,1229 @@
+// Copyright (c) 2023 IndyKite
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package authorization_test
+
+import (
+	"context"
+
+	"github.com/fatih/structs"
+	"github.com/grpc-ecosystem/go-grpc-middleware/v2/interceptors/retry"
+
+	authorizationpb "github.com/indykite/indykite-sdk-go/gen/indykite/authorization/v1beta1"
+	objectpb "github.com/indykite/indykite-sdk-go/gen/indykite/objects/v1beta1"
+	integration "github.com/indykite/indykite-sdk-go/test"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	. "github.com/onsi/gomega/gstruct"
+)
+
+var _ = Describe("Authorized", func() {
+	Describe("IsAuthorized", func() {
+		It("IsAuthorizedDT", func() {
+			var err error
+			authorizationClient, err := integration.InitConfigAuthorization()
+			Expect(err).To(Succeed())
+
+			digitalTwin := &authorizationpb.DigitalTwin{
+				Id: integration.DigitalTwin1,
+			}
+
+			resources := integration.Resource1
+			inputParams := map[string]*authorizationpb.InputParam{}
+			var policyTags []string
+
+			resp, err := authorizationClient.IsAuthorized(
+				context.Background(),
+				digitalTwin,
+				resources,
+				inputParams,
+				policyTags,
+				retry.WithMax(2),
+			)
+
+			Expect(err).To(Succeed())
+			Expect(resp).NotTo(BeNil())
+
+			decision := resources[0].Type
+			resource := resources[0].ExternalId
+			action := resources[0].Actions[0]
+
+			newMap := structs.Map(resp)
+			Expect(newMap).To(MatchAllKeys(Keys{
```

Comment:
Why you convert Struct to Map? Please write a matcher directly for the given struct and do not bring any new dependencies unless they are really needed.